arrow/request_test.go

```diff
@@ -0,0 +1,154 @@
+package arrow_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tarantool/go-iproto"
+	"github.com/tarantool/go-tarantool/v2"
+	"github.com/tarantool/go-tarantool/v2/arrow"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+// INSERT Arrow request.
+//
+// FIXME: replace with iproto.IPROTO_INSERT_ARROW when iproto will released.
+// https://github.com/tarantool/go-tarantool/issues/412
+const iprotoInsertArrowType = iproto.Type(17)
+
+const validSpace uint32 = 1 // Any valid value != default.
+
+func TestInsertRequestType(t *testing.T) {
+	request := arrow.NewInsertRequest(validSpace)
+	require.Equal(t, iprotoInsertArrowType, request.Type())
+}
+
+func TestInsertRequestAsync(t *testing.T) {
+	request := arrow.NewInsertRequest(validSpace)
+	require.Equal(t, false, request.Async())
+}
+
+func TestInsertRequestCtx_default(t *testing.T) {
+	request := arrow.NewInsertRequest(validSpace)
+	require.Equal(t, nil, request.Ctx())
+}
+
+func TestInsertRequestCtx_setter(t *testing.T) {
+	ctx := context.Background()
+	request := arrow.NewInsertRequest(validSpace).Context(ctx)
+	require.Equal(t, ctx, request.Ctx())
+}
+
+func TestResponseDecode(t *testing.T) {
+	header := tarantool.Header{}
+	baseResponse, err := tarantool.DecodeBaseResponse(header, bytes.NewBuffer([]byte{}))
+	require.NoError(t, err)
+
+	buf := bytes.NewBuffer([]byte{})
+	enc := msgpack.NewEncoder(buf)
+
+	enc.EncodeMapLen(1)
+	enc.EncodeUint8(uint8(iproto.IPROTO_DATA))
+	enc.Encode([]interface{}{'v', '2'})
+
+	request := arrow.NewInsertRequest(validSpace)
+	resp, err := request.Response(header, bytes.NewBuffer(buf.Bytes()))
+	require.NoError(t, err)
+	require.IsType(t, baseResponse, resp)
+	require.Equal(t, header, resp.Header())
+
+	decodedInterface, err := resp.Decode()
+	require.NoError(t, err)
+	require.Equal(t, []interface{}{'v', '2'}, decodedInterface)
+}
+
+func TestResponseDecodeTyped(t *testing.T) {
+	header := tarantool.Header{}
+	baseResponse, err := tarantool.DecodeBaseResponse(header, bytes.NewBuffer([]byte{}))
+	require.NoError(t, err)
+
+	buf := bytes.NewBuffer([]byte{})
+	enc := msgpack.NewEncoder(buf)
+
+	enc.EncodeMapLen(1)
+	enc.EncodeUint8(uint8(iproto.IPROTO_DATA))
+	enc.EncodeBytes([]byte{'v', '2'})
+
+	request := arrow.NewInsertRequest(validSpace)
+	resp, err := request.Response(header, bytes.NewBuffer(buf.Bytes()))
+	require.NoError(t, err)
+	require.IsType(t, baseResponse, resp)
```

Comment:
Let's do without this check. It tells too much of the internal implementation and will complicate refactoring in the future.

Reply:
Here is just an improved version of
```Go
assert.True(t, fmt.Sprintf("%T", resp) == fmt.Sprintf("%T", test.expected))
```
From the similar `go-tarantool/request_test.go` tests solution.
Do you think such a test is unnecessary?

Reply:
Done.